### PR TITLE
DOC-767 Add secret management to RPCN in Cloud

### PR DESCRIPTION
**Pull request by asimms41**
## Description

Resolves https://github.com/redpanda-data/documentation-private/issues/<add-your-issue-number-here>
Review deadline: 29 November

## Page previews

[Secrets](https://deploy-preview-138--rp-cloud.netlify.app/redpanda-cloud/security/secrets/)
[Secret management](https://deploy-preview-138--rp-cloud.netlify.app/redpanda-cloud/develop/connect/configuration/secret-management/)

NOTE: I will raise a separate PR to update the What's new in Cloud.

## Checks

- [x] New feature
- [ ] Content gap
- [ ] Support Follow-up
- [ ] Small fix (typos, links, copyedits, etc)

### CODE REVIEW

File: modules/develop/pages/connect/configuration/secret-management.adoc [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+
+You can manage secrets from the Cloud UI or Data Plane API.
+
+=== Create a secret
+
+You can create a secret and reference it in multiple data pipelines on the same cluster.
+
+[tabs]
+=====
+Cloud UI (BYOC and Dedicated)::
```
Comment (asimms41) [inline modules/develop/pages/connect/configuration/secret-management.adoc:23]: This assumes that Dedicated will be announced at the same time as secret management. If that is not correct, I'll remove Dedicated from this tab.

File: modules/security/pages/secrets.adoc [hunk 2]
```diff
@@ -7,8 +7,7 @@ have policies defined by the actions and resources that a user (also
 known as a principal) strictly needs, following the principle of least privilege.
 
 Redpanda Cloud also uses _static secrets_, stored in either the
-https://aws.amazon.com/secrets-manager/[AWS Secrets Manager^] or
-https://cloud.google.com/secret-manager[GCP Secret Manager^] services. Static
+https://aws.amazon.com/secrets-manager/[AWS Secrets Manager^], https://cloud.google.com/secret-manager[GCP Secret Manager^] services or https://azure.microsoft.com/en-us/products/key-vault[Azure Key Vault^]. Static
```
Comment (asimms41) [inline modules/security/pages/secrets.adoc:10]: Is this correct?

File: modules/develop/pages/connect/configuration/secret-management.adoc [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+:description: Learn how to manage secrets in Redpanda Connect using the Cloud UI or Data Plane API, and how to add them to your data pipelines.
+
+Learn how to manage secrets in Redpanda Connect, and how to add them to your data pipelines without exposing them.
+
+Secrets are stored in the secret management solution of your Cloud provider and are retrieved when you run a pipeline configuration that references them. For more information about the way secrets are stored in Redpanda Cloud, see xref:security:secrets.adoc[Secrets].
+
+== Prerequisites
+
+* A running BYOC, Dedicated or Serverless cluster
+* Secrets management enabled
```
Comment (asimms41) [inline modules/develop/pages/connect/configuration/secret-management.adoc:11]: Do users need to do anything to get secrets management enabled?

File: modules/develop/pages/connect/configuration/secret-management.adoc [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+ -H 'accept: application/json'\
+ -H 'authorization: Bearer <token>'\
+----
++
+You must include the following values:
+
+- `<dataplane-api-url>`: The base URL for the Data Plane API.
+- `<secret-name>`: The name of the secret you want to delete.
+- `<token>`: The API key you generated during authentication.
+--
```
Comment (asimms41) [inline modules/develop/pages/connect/configuration/secret-management.adoc:198]: In curl, I didn't a response....Is that correct?

File: modules/develop/pages/connect/configuration/secret-management.adoc [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+The response returns the name of the secret and the scope `"SCOPE_REDPANDA_CONNECT"`.
+
+--
+=====
+
+=== Delete a secret
+
+Before you delete a secret, make sure that you remove references to it from your data pipelines. 
+
+NOTE: Changes do not affect pipelines that are already running. Secrets are not removed until a pipeline is restarted.
```
Comment (tomasz-sadura) [inline modules/develop/pages/connect/configuration/secret-management.adoc:152]: `Secrets are not removed until a pipeline is restarted.` - not sure what you mean exactly here, as I would say the secret is actually removed right away. `Changes do not affect pipelines that are already running.` is true.

File: modules/develop/pages/connect/configuration/secret-management.adoc (hunk 4 above)
Comment (tomasz-sadura) [reply modules/develop/pages/connect/configuration/secret-management.adoc:198]: yes, there is not body in the response if the secret existed and was deleted

File: modules/develop/pages/connect/configuration/secret-management.adoc (hunk 5 above)
Comment (asimms41) [reply modules/develop/pages/connect/configuration/secret-management.adoc:159]: Updated.

File: modules/security/pages/secrets.adoc (hunk 2 above)
Comment (asimms41) [reply modules/security/pages/secrets.adoc]: I have removed reference to this page as it doesn't add anything to the explanation.

File: modules/develop/pages/connect/configuration/secret-management.adoc (hunk 3 above)
Comment (asimms41) [reply modules/develop/pages/connect/configuration/secret-management.adoc]: Updated.

File: modules/develop/pages/connect/configuration/secret-management.adoc [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+
+* A running BYOC, Dedicated or Serverless cluster
+* On BYOC clusters only, check that secrets management is enabled:
+
+.. Log in to https://cloud.redpanda.com[Redpanda Cloud^].
+.. Go to the **Connect** page.
+.. Select the **Redpanda Connect** tab and make sure you can see a **Secrets** tab.
+
++
+If you cannot a **Secrets** tab, contact https://support.redpanda.com/hc/en-us/requests/new[Redpanda Support^]
```
Comment (nicolaferraro) [inline modules/develop/pages/connect/configuration/secret-management.adoc:18]: missing verb

**Review by tomasz-sadura — APPROVED** [review]

File: modules/develop/pages/connect/configuration/secret-management.adoc (hunk 6 above)
Comment (asimms41) [reply modules/develop/pages/connect/configuration/secret-management.adoc]: Updated

**Review by JakeSCahill — APPROVED** [review]
Awesome feature and great doc 👍

**Pull request merged**